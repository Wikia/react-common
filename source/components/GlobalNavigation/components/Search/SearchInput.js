import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import Icon from '../../../Icon';
import Button from '../../../Button';

import I18nNamespaceContext from '../../context/I18nNamespaceContext';

const SearchInput = React.forwardRef((props, ref) => {
    const {
        model,
        query,
        onSearchActivation,
        onQueryChanged,
        onFocus,
        onBlur,
        onSearchClear,
        onSearchClose,
        onSearchSubmit,
    } = props;
    const placeholderConfig = model['placeholder-active'];
    const [t] = useTranslation(useContext(I18nNamespaceContext));

    return (
        <React.Fragment>
            <input
                className="wds-global-navigation__search-input"
                placeholder={t(placeholderConfig.key, { sitename: placeholderConfig.params.sitename.value })}
                value={query}
                onClick={onSearchActivation}
                ref={ref}
                onChange={onQueryChanged}
                onFocus={onFocus}
                onBlur={onBlur}
                type="search"
            />
            <Button
                className="wds-global-navigation__search-clear"
                type="button"
                onClick={onSearchClear}
                text
            >
                <Icon name="add" small className="wds-global-navigation__search-clear-icon" />
            </Button>
            <Button
                className="wds-global-navigation__search-close"
                type="button"
                onClick={onSearchClose}
                text
            >
                <Icon name="close-tiny" className="wds-global-navigation__search-close-icon" tiny />
            </Button>
            <Button
                className="wds-global-navigation__search-submit"
                type="submit"
                disabled={!query}
                onClick={onSearchSubmit}
                data-tracking-label={model.results['tracking-label']}
            >
                <Icon name="arrow-small" className="wds-global-navigation__search-submit-icon" small />
            </Button>
        </React.Fragment>
    );
});

SearchInput.propTypes = {
    model: PropTypes.shape().isRequired,
    onBlur: PropTypes.func.isRequired,
    onFocus: PropTypes.func.isRequired,
    onQueryChanged: PropTypes.func.isRequired,
    onSearchActivation: PropTypes.func.isRequired,
    onSearchClear: PropTypes.func.isRequired,
    onSearchClose: PropTypes.func.isRequired,
    onSearchSubmit: PropTypes.func.isRequired,
    query: PropTypes.string.isRequired,
};

export default SearchInput;
