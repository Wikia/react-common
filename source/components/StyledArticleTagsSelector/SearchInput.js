import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

import IconMagnifyingGlassSmall from '../../icons/IconMagnifyingGlassSmall';
import IconCloseSmall from '../../icons/IconCloseSmall';

import { colorWithAlpha } from '../../utils/css';

import StyledList from '../StyledList';
import HighlightedText from '../HighlightedText';

import TagShape from './TagShape';

const ICON_STYLES = css`
    fill: currentColor;
    height: 16px;
    width: 16px;
`;

const List = styled(StyledList)`
    background-color: ${({ theme }) => theme.color.white};
    border: 1px solid ${({ theme }) => theme.color.fandom_middle_gray};
    overflow: hidden;
    padding: 10px 0;
    position: absolute;
    width: 100%;
    top: 32px;

    @media ${({ theme }) => theme.media.medium_up} {
        top: 24px;
    }

    > li {
        cursor: pointer;
        line-height: 36px;
        margin: 0 10px;
        padding: 0 10px;

        &:hover {
            background-color: ${({ accentColor }) => colorWithAlpha(accentColor, 0.1)};
        }
    }

    mark {
        background-color: transparent;
        font-weight: ${({ theme }) => theme.font_weight.bold};
    }
`;

const Slider = styled.div`
    @media ${({ theme }) => theme.media.small_down} {
        background-color: ${({ theme }) => theme.color.white};
        height: 100vh;
        left: 0;
        position: fixed;
        top: 0;
        width: 100vw;
        z-index: 1;
    }

    @media ${({ theme }) => theme.media.medium_up} {
        flex: 1;
    }
`;

const Wrapper = styled.div`
    align-items: center;
    border-bottom: solid 1px ${({ theme }) => theme.color.black};
    display: flex;
    position: relative;

    @media ${({ theme }) => theme.media.small_only} {
        margin: 13px;
    }
`;

const Input = styled.input`
    background-color: transparent;
    border: none;
    flex-grow: 1;
    font-size: ${({ theme }) => theme.font_size.base};
    line-height: ${({ theme }) => theme.line_height.normal};
    width: 100%;

    @media ${({ theme }) => theme.media.medium_up} {
        font-size: ${({ theme }) => theme.font_size.s};
    }

    &:focus {
        outline: none;
    }
`;

const IconSearch = styled(IconMagnifyingGlassSmall)`
    ${ICON_STYLES}
    margin-right: 6px;
`;

const IconClose = styled(IconCloseSmall)`
    ${ICON_STYLES}

    cursor: pointer;
`;

function SearchInput({
    accentColor,
    className,
    onChange,
    onClose,
    onAddTag,
    communityName,
    list,
    query,
}) {
    const ref = React.createRef();

    return (
        <Slider className={className}>
            <Wrapper>
                <IconSearch />
                <Input
                    placeholder={`Searching ${communityName}`}
                    ref={ref}
                    onChange={onChange}
                    autoFocus
                />
                {list && (
                    <List accentColor={accentColor}>
                        {list.map(({ id, title }) => (
                            <li
                                key={id}
                                onClick={/* istanbul ignore next */ () => onAddTag(id)}
                            >
                                <HighlightedText
                                    highlight={query}
                                    text={title}
                                />
                            </li>
                        ))}
                    </List>
                )}
                <IconClose onClick={onClose} />
            </Wrapper>
        </Slider>
    );
}

SearchInput.propTypes = {
    accentColor: PropTypes.string.isRequired,
    /** Extra class name */
    className: PropTypes.string,
    communityName: PropTypes.string,
    list: PropTypes.arrayOf(TagShape),
    onAddTag: PropTypes.func,
    onChange: PropTypes.func,
    onClose: PropTypes.func,
    query: PropTypes.string,
};

SearchInput.defaultProps = {
    className: '',
    communityName: '',
    list: null,
    onAddTag: noop,
    onChange: noop,
    onClose: noop,
    query: '',
};

export default SearchInput;
