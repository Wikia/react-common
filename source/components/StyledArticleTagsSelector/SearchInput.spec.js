import React from 'react';

import SearchInput from './SearchInput';

const tags = [
    { id: '1', title: 'Foo' },
    { id: '2', title: 'Bar' },
    { id: '3', title: 'FooBar' },
];

test('SearchInput renders empty', () => {
    const component = mountWithThemeProvider(<SearchInput />);
    expect(component.find(SearchInput)).toMatchSnapshot();
});


test('SearchInput renders with props', () => {
    const component = mountWithThemeProvider(<SearchInput
        className="foo"
        communityName="bar"
        query="q"
        list={tags}
    />);
    expect(component.find(SearchInput)).toMatchSnapshot();
});
