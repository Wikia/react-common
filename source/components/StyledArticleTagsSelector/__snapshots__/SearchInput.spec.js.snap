// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders empty 1`] = `
<SearchInput
  className=""
  communityName=""
  list={null}
  onAddTag={[Function]}
  onChange={[Function]}
  onClose={[Function]}
  query=""
>
  <SearchInput__Wrapper
    className=""
  >
    <StyledComponent
      className=""
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "SearchInput__Wrapper-sc-1cm9pl1-1",
            "isStatic": false,
            "lastClassName": "bBJnHG",
            "rules": Array [
              "align-items:center;border-bottom:solid 1px ",
              [Function],
              ";display:flex;position:relative;@media ",
              [Function],
              "{}@media ",
              [Function],
              "{flex:1;}",
            ],
          },
          "displayName": "SearchInput__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "SearchInput__Wrapper-sc-1cm9pl1-1",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="SearchInput__Wrapper-sc-1cm9pl1-1 bBJnHG"
      >
        <SearchInput__IconSearch>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__IconSearch-sc-1cm9pl1-3",
                  "isStatic": true,
                  "lastClassName": "qVP",
                  "rules": Array [
                    "fill:currentColor;height:16px;width:16px;",
                    " margin-right:6px;",
                  ],
                },
                "displayName": "SearchInput__IconSearch",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__IconSearch-sc-1cm9pl1-3",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <_default
              className="SearchInput__IconSearch-sc-1cm9pl1-3 qVP"
            >
              <_default
                className="SearchInput__IconSearch-sc-1cm9pl1-3 qVP"
              >
                <div
                  className="SearchInput__IconSearch-sc-1cm9pl1-3 qVP"
                />
              </_default>
            </_default>
          </StyledComponent>
        </SearchInput__IconSearch>
        <SearchInput__Input
          autoFocus={true}
          onChange={[Function]}
          placeholder="Searching "
        >
          <StyledComponent
            autoFocus={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__Input-sc-1cm9pl1-2",
                  "isStatic": false,
                  "lastClassName": "ffxNmN",
                  "rules": Array [
                    "border:none;line-height:",
                    [Function],
                    ";width:100%;background-color:transparent;flex-grow:1;@media ",
                    [Function],
                    "{font-size:",
                    [Function],
                    ";}&:focus{outline:none;}",
                  ],
                },
                "displayName": "SearchInput__Input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__Input-sc-1cm9pl1-2",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={
              Object {
                "current": <input
                  class="SearchInput__Input-sc-1cm9pl1-2 ffxNmN"
                  placeholder="Searching "
                />,
              }
            }
            onChange={[Function]}
            placeholder="Searching "
          >
            <input
              autoFocus={true}
              className="SearchInput__Input-sc-1cm9pl1-2 ffxNmN"
              onChange={[Function]}
              placeholder="Searching "
            />
          </StyledComponent>
        </SearchInput__Input>
        <SearchInput__IconClose
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__IconClose-sc-1cm9pl1-4",
                  "isStatic": true,
                  "lastClassName": "hSmGuN",
                  "rules": Array [
                    "fill:currentColor;height:16px;width:16px;",
                    " cursor:pointer;",
                  ],
                },
                "displayName": "SearchInput__IconClose",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__IconClose-sc-1cm9pl1-4",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <_default
              className="SearchInput__IconClose-sc-1cm9pl1-4 hSmGuN"
              onClick={[Function]}
            >
              <_default
                className="SearchInput__IconClose-sc-1cm9pl1-4 hSmGuN"
                onClick={[Function]}
              >
                <div
                  className="SearchInput__IconClose-sc-1cm9pl1-4 hSmGuN"
                  onClick={[Function]}
                />
              </_default>
            </_default>
          </StyledComponent>
        </SearchInput__IconClose>
      </div>
    </StyledComponent>
  </SearchInput__Wrapper>
</SearchInput>
`;

exports[`SearchInput renders with props 1`] = `
<SearchInput
  accentColor="#00f"
  className="foo"
  communityName="bar"
  list={
    Array [
      Object {
        "id": "1",
        "title": "Foo",
      },
      Object {
        "id": "2",
        "title": "Bar",
      },
      Object {
        "id": "3",
        "title": "FooBar",
      },
    ]
  }
  onAddTag={[Function]}
  onChange={[Function]}
  onClose={[Function]}
  query="q"
>
  <SearchInput__Wrapper
    className="foo"
  >
    <StyledComponent
      className="foo"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "SearchInput__Wrapper-sc-1cm9pl1-1",
            "isStatic": false,
            "lastClassName": "bBJnHG",
            "rules": Array [
              "align-items:center;border-bottom:solid 1px ",
              [Function],
              ";display:flex;position:relative;@media ",
              [Function],
              "{}@media ",
              [Function],
              "{flex:1;}",
            ],
          },
          "displayName": "SearchInput__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "SearchInput__Wrapper-sc-1cm9pl1-1",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="foo SearchInput__Wrapper-sc-1cm9pl1-1 bBJnHG"
      >
        <SearchInput__IconSearch>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__IconSearch-sc-1cm9pl1-3",
                  "isStatic": true,
                  "lastClassName": "qVP",
                  "rules": Array [
                    "fill:currentColor;height:16px;width:16px;",
                    " margin-right:6px;",
                  ],
                },
                "displayName": "SearchInput__IconSearch",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__IconSearch-sc-1cm9pl1-3",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <_default
              className="SearchInput__IconSearch-sc-1cm9pl1-3 qVP"
            >
              <_default
                className="SearchInput__IconSearch-sc-1cm9pl1-3 qVP"
              >
                <div
                  className="SearchInput__IconSearch-sc-1cm9pl1-3 qVP"
                />
              </_default>
            </_default>
          </StyledComponent>
        </SearchInput__IconSearch>
        <SearchInput__Input
          autoFocus={true}
          onChange={[Function]}
          placeholder="Searching bar"
        >
          <StyledComponent
            autoFocus={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__Input-sc-1cm9pl1-2",
                  "isStatic": false,
                  "lastClassName": "ffxNmN",
                  "rules": Array [
                    "border:none;line-height:",
                    [Function],
                    ";width:100%;background-color:transparent;flex-grow:1;@media ",
                    [Function],
                    "{font-size:",
                    [Function],
                    ";}&:focus{outline:none;}",
                  ],
                },
                "displayName": "SearchInput__Input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__Input-sc-1cm9pl1-2",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={
              Object {
                "current": <input
                  class="SearchInput__Input-sc-1cm9pl1-2 ffxNmN"
                  placeholder="Searching bar"
                />,
              }
            }
            onChange={[Function]}
            placeholder="Searching bar"
          >
            <input
              autoFocus={true}
              className="SearchInput__Input-sc-1cm9pl1-2 ffxNmN"
              onChange={[Function]}
              placeholder="Searching bar"
            />
          </StyledComponent>
        </SearchInput__Input>
        <SearchInput__List
          accentColor="#00f"
          bigItems={false}
          boldItems={false}
          className=""
          hasEllipsis={false}
          isLinked={false}
          linesBetween={false}
        >
          <StyledComponent
            accentColor="#00f"
            bigItems={false}
            boldItems={false}
            className=""
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__List-sc-1cm9pl1-0",
                  "isStatic": false,
                  "lastClassName": "cveBSp",
                  "rules": Array [
                    "list-style-type:none;margin:0;padding:7px 18px;",
                    [Function],
                    " > li{font-size:",
                    [Function],
                    ";font-weight:normal;line-height:1em;padding:",
                    "11",
                    "px 0;",
                    [Function],
                    " ",
                    [Function],
                    " ",
                    [Function],
                    " ",
                    [Function],
                    " ",
                    [Function],
                    "}",
                    "background-color:",
                    [Function],
                    ";border:1px solid ",
                    [Function],
                    ";overflow:hidden;padding:10px 0;position:absolute;width:100%;@media ",
                    [Function],
                    "{top:24px;}> li{cursor:pointer;line-height:36px;margin:0 10px;padding:0 10px;&:hover{background-color:",
                    [Function],
                    ";}}mark{background-color:transparent;font-weight:",
                    [Function],
                    ";}",
                  ],
                },
                "defaultProps": Object {
                  "bigItems": false,
                  "boldItems": false,
                  "children": null,
                  "className": "",
                  "hasEllipsis": false,
                  "isLinked": false,
                  "linesBetween": false,
                },
                "displayName": "SearchInput__List",
                "foldedComponentIds": Array [
                  "StyledList-sc-1t1k18i-0",
                ],
                "propTypes": Object {
                  "bigItems": [Function],
                  "boldItems": [Function],
                  "children": [Function],
                  "className": [Function],
                  "hasEllipsis": [Function],
                  "isLinked": [Function],
                  "linesBetween": [Function],
                },
                "render": [Function],
                "styledComponentId": "SearchInput__List-sc-1cm9pl1-0",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            hasEllipsis={false}
            isLinked={false}
            linesBetween={false}
          >
            <ul
              className="StyledList-sc-1t1k18i-0 SearchInput__List-sc-1cm9pl1-0 cveBSp"
            >
              <li
                key="1"
                onClick={[Function]}
              >
                <HighlightedText
                  highlight="q"
                  text="Foo"
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Foo",
                      }
                    }
                  />
                </HighlightedText>
              </li>
              <li
                key="2"
                onClick={[Function]}
              >
                <HighlightedText
                  highlight="q"
                  text="Bar"
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Bar",
                      }
                    }
                  />
                </HighlightedText>
              </li>
              <li
                key="3"
                onClick={[Function]}
              >
                <HighlightedText
                  highlight="q"
                  text="FooBar"
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "FooBar",
                      }
                    }
                  />
                </HighlightedText>
              </li>
            </ul>
          </StyledComponent>
        </SearchInput__List>
        <SearchInput__IconClose
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__IconClose-sc-1cm9pl1-4",
                  "isStatic": true,
                  "lastClassName": "hSmGuN",
                  "rules": Array [
                    "fill:currentColor;height:16px;width:16px;",
                    " cursor:pointer;",
                  ],
                },
                "displayName": "SearchInput__IconClose",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__IconClose-sc-1cm9pl1-4",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <_default
              className="SearchInput__IconClose-sc-1cm9pl1-4 hSmGuN"
              onClick={[Function]}
            >
              <_default
                className="SearchInput__IconClose-sc-1cm9pl1-4 hSmGuN"
                onClick={[Function]}
              >
                <div
                  className="SearchInput__IconClose-sc-1cm9pl1-4 hSmGuN"
                  onClick={[Function]}
                />
              </_default>
            </_default>
          </StyledComponent>
        </SearchInput__IconClose>
      </div>
    </StyledComponent>
  </SearchInput__Wrapper>
</SearchInput>
`;
